src/main/java/utils/configurations/Configurations.java

```diff
@@ -0,0 +1,36 @@
+package utils.configurations;
+
+import com.google.common.collect.ImmutableMap;
+import onlineservices.Service;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import services.WindowControl.WindowControlService;
+import utils.filereaders.CsvFileReader;
+
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+@Configuration
+public class Configurations {
+    private final ImmutableMap<String, Service> allServices = ImmutableMap.of(
```

Comment:
my mistake, this will not work because there is no guarantee that the service wont be created multiple times, remove the map, add all the services as beans:
@Bean
public WindowControlOnlineService windowControlOnlineService() {
    return new WindowControlOnlineService();
}
this way spring will ensure that every time the WindowControlOnlineService is requested, the same instance will be used